Description: Ensure that -momit-leaf-frame-pointer is only used on architectures that support it
Author: James Page <james.page@ubuntu.com>
Forwarded: https://github.com/facebook/rocksdb/pull/1019

--- a/src/rocksdb/Makefile
+++ b/src/rocksdb/Makefile
@@ -84,7 +84,8 @@ endif
 # compile with -O2 if debug level is not 2
 ifneq ($(DEBUG_LEVEL), 2)
 OPT += -O2 -fno-omit-frame-pointer
-ifneq ($(MACHINE),ppc64) # ppc64 doesn't support -momit-leaf-frame-pointer
+# Skip for archs that don't support -momit-leaf-frame-pointer
+ifeq (,$(shell $(CXX) -fsyntax-only -momit-leaf-frame-pointer -xc /dev/null 2>&1))
 OPT += -momit-leaf-frame-pointer
 endif
 endif
